kotlinx.coroutines.JobCancellationException: BlockingCoroutine is cancelling

### Is there an existing issue for this?

- [X] I have searched the existing issues and didn't find mine.

### Steps to reproduce

Create a flow with a name that includes a `/` (on an operating system that uses `/` as a path separator`), for example:

```yaml
appId: your.app.id
name: foo / bar
---
- launchApp
```

### Actual results

The flow appears to run, but `maestro` errors with an exception as soon as the flow finishes:

```
❌ Error: BlockingCoroutine is cancelling
kotlinx.coroutines.JobCancellationException: BlockingCoroutine is cancelling; job=BlockingCoroutine{Cancelling}@1ef0ac30
Caused by: java.io.FileNotFoundException: /Users/jonathan/.maestro/tests/2024-09-05_085603/ai-report-foo / bar.html (No such file or directory)
	at java.base/java.io.FileOutputStream.open0(Native Method)
	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:184)
	at kotlin.io.FilesKt__FileReadWriteKt.writeBytes(FileReadWrite.kt:108)
	at kotlin.io.FilesKt__FileReadWriteKt.writeText(FileReadWrite.kt:134)
	at kotlin.io.FilesKt__FileReadWriteKt.writeText$default(FileReadWrite.kt:134)
	at maestro.cli.report.HtmlAITestSuiteReporter.report(HtmlAITestSuiteReporter.kt:47)
	at maestro.cli.report.TestDebugReporter.saveSuggestions(TestDebugReporter.kt:64)
	at maestro.cli.runner.TestRunner.runSingle(TestRunner.kt:76)
	at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1$1.invoke(TestCommand.kt:306)
	at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1$1.invoke(TestCommand.kt:258)
	at maestro.cli.session.MaestroSessionManager.newSession(MaestroSessionManager.kt:102)
	at maestro.cli.session.MaestroSessionManager.newSession$default(MaestroSessionManager.kt:54)
	at maestro.cli.command.TestCommand$handleSessions$1$1$results$1$1.invokeSuspend(TestCommand.kt:258)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
```

The caused by error contains the hint:
```
Caused by: java.io.FileNotFoundException: /Users/jonathan/.maestro/tests/2024-09-05_085603/ai-report-foo / bar.html 
```

So it seems to be trying to write to `ai-report-${FLOW_NAME}.html` which ends up trying to write to a file called `bar.html` in a directory that doesn't exist.

### Expected results

Operations that write to files with user-given input (such as the flow name) should escape/replace characters that have special meaning in filenames, such as the path separator and `..` too.

### About app

N/A

### About environment

- openjdk version "17.0.12" 2024-07-16 LTS
- macOS 14.5
- arm64

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. *DO NOT* remove the backticks! -->
```

</details>


### Maestro version

1.38.1

### How did you install Maestro?

install script (https://get.maestro.mobile.dev)

### Anything else?

_No response_

--- a/maestro-cli/src/main/java/maestro/cli/report/HtmlAITestSuiteReporter.kt
+++ b/maestro-cli/src/main/java/maestro/cli/report/HtmlAITestSuiteReporter.kt
@@ -27,7 +27,10 @@ import java.io.File
 class HtmlAITestSuiteReporter {
 
     private val FlowAIOutput.htmlReportFilename
-        get() = "ai-report-${flowName}.html"
+        // Sanitize the flow name so a `/` (or other path separator) in it does not
+        // turn the report filename into a path pointing at a non-existent directory,
+        // which would crash with FileNotFoundException. See issue #2017.
+        get() = "ai-report-${flowName.replace("/", "_")}.html"
 
     private val reportCss: String
         get() = readResourceAsText(this::class, "/ai_report.css")

--- a/maestro-cli/src/test/kotlin/maestro/cli/report/HtmlAITestSuiteReporterTest.kt
+++ b/maestro-cli/src/test/kotlin/maestro/cli/report/HtmlAITestSuiteReporterTest.kt
@@ -1,0 +1,29 @@
+package maestro.cli.report
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+class HtmlAITestSuiteReporterTest {
+
+    @Test
+    fun `report does not crash when the flow name contains a path separator`(@TempDir tempDir: File) {
+        // Given - a flow whose name contains a '/', which is a path separator on unix-like systems.
+        // See issue #2017: writing "ai-report-foo / bar.html" used to be interpreted as a path into
+        // a non-existent "foo " directory and crashed with FileNotFoundException.
+        val reporter = HtmlAITestSuiteReporter()
+        val output = FlowAIOutput(
+            flowName = "foo / bar",
+            flowFile = File(tempDir, "flow.yaml").apply { writeText("appId: com.example") },
+            screenOutputs = mutableListOf(),
+        )
+
+        // When
+        reporter.report(listOf(output), tempDir)
+
+        // Then - the report is written to a single, sanitized file inside the output directory.
+        val reportFile = File(tempDir, "ai-report-foo _ bar.html")
+        assertThat(reportFile.exists()).isTrue()
+    }
+}

